conans/model/conf.py

```diff
@@ -481,7 +481,7 @@ def copy_conaninfo_conf(self):
         # Reading the list of all the configurations selected by the user to use for the package_id
         package_id_confs = self.get("tools.info.package_id:confs", default=[], check_type=list)
         for conf_name in package_id_confs:
-            matching_confs = [c for c in self._values if re.match(conf_name, c)] or [conf_name]
+            matching_confs = [c for c in self._values if re.match(conf_name, c)]
```

Comment:
The original idea from @thorsten-klein seems to have been that if no match was found, then the passed conf would be a specific conf id - This is not the case for patterns that just don't match anything, either because they are malformed if they are meant to add built in confs, or because they target user. confs that are not currently defined.

I don't see a need to keep the or branch as proper confs will only match themselves